### PR TITLE
8264872: Dependencies: Migrate to PerfData counters

### PR DESCRIPTION
**Pull request by iwanowww**
Migrate performance counters on `Dependencies` (`deps_find_witness_*`) to `PerfData`.

Testing:
- [x] hs-tier1 - hs-tier2

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264872](https://bugs.openjdk.java.net/browse/JDK-8264872): Dependencies: Migrate to PerfData counters

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 3567e0f069d8e91ea8e168330c318a7aed0f219b
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**) ⚠️ Review applies to 3567e0f069d8e91ea8e168330c318a7aed0f219b

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3386/head:pull/3386` \
`$ git checkout pull/3386`

Update a local copy of the PR: \
`$ git checkout pull/3386` \
`$ git pull https://git.openjdk.java.net/jdk pull/3386/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3386`

View PR using the GUI difftool: \
`$ git pr show -t 3386`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3386.diff">https://git.openjdk.java.net/jdk/pull/3386.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

File: src/hotspot/share/code/dependencies.cpp [hunk 1]
```diff
@@ -1179,53 +1198,32 @@ class ClassHierarchyWalker {
...
-                      "calls=%d, steps=%d (avg=%.1f), recursions=%d, singles=%d",
-                      deps_find_witness_calls,
-                      deps_find_witness_steps,
-                      (double)deps_find_witness_steps / deps_find_witness_calls,
-                      deps_find_witness_recursions,
-                      deps_find_witness_singles);
-      }
-    }
-    return true;
+PerfCounter* ClassHierarchyWalker::_perf_find_witness_anywhere_calls_count   = NULL;
```
Comment (neliasso) [inline src/hotspot/share/code/dependencies.cpp:1212]: Align '=' on all three or skip extra space

**Review by neliasso — APPROVED** [review]
Looks good.

File: src/hotspot/share/code/dependencies.cpp (hunk 1 above)
Comment (neliasso) [reply src/hotspot/share/code/dependencies.cpp:1212]: I'm refering to the whitespace before '= NULL'

**Comment by iwanowww** [comment]
Thanks for the reviews, Vladimir and Nils.

**Pull request closed without merging**